src/adapter/cloudflare-pages/handler.ts

```diff
@@ -34,6 +43,24 @@ export const handle: HandleInterface = (app: Hono) => (eventContext: EventContex
   )
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function handleMiddleware<E extends Env = any, P extends string = any, I extends Input = {}>(
+  middleware: MiddlewareHandler<E, P, I>
+): PagesFunction {
+  return async (executionCtx) => {
+    const context = new Context(new HonoRequest(executionCtx.request), {
+      env: executionCtx.env,
+      executionCtx,
+    })
+
+    const response = await middleware(context, async () => {
+      context.res = await executionCtx.next()
+    })
+
+    return response ?? context.res
+  }
```

Comment:
This is a very naive implementation, I haven't spent much time looking at the internals of hono yet. Appreciate any feedback on how this might be improved